src/parse.c

```diff
@@ -1105,12 +1105,76 @@ handle_addresses(yaml_document_t* doc, yaml_node_t* node, const void* _, GError*
     return handle_generic_addresses(doc, node, TRUE, &(cur_netdef->ip4_addresses), &(cur_netdef->ip6_addresses), error);
 }
 
+const static char *GW_MULTIPLE_IP_ERROR = "gateway: there must be at most one of each IP type";
+
+struct gw_parse_result {
```

Comment:
We should probably prefix this with "_" (`_gw_parse_result`) to mark it as: "for internal use only" and maybe move it into the `parse.h` header.